toolbox/message_tools/message_maker.py

```diff
@@ -122,16 +124,98 @@ def create_refusal_message():
         print('Successfully published')
 
 
+def check_uac_qid_pair_exist(uac, qid):
+    result = execute_parametrized_sql_query("SELECT qid FROM uacqid.uac_qid WHERE uac = %s", (uac,))
+    if not result:
+        raise ValueError('ERROR: UAC not found in uac qid database')
+
+    found_qid = result[0][0]
+    if found_qid != qid:
+        raise ValueError(f'ERROR: UAC exists but is linked to a different QID: {found_qid}')
+
+
+def create_rm_uac_created_message():
+    print('!!! WARNING !!!')
+    print('You must be absolutely sure this UAC/QID pair is genuine, it MUST exist in the uacqid service database '
+          'but NOT already exist in our UAC QID link table')
+    speed_bump_response = input('Are you sure you wish to proceed? (y/n): ')
+    if speed_bump_response.lower() not in {'yes', 'y'}:
+        return
+    print('Ok, generating an RM UAC CREATED message.')
+    print('Whats the QID?')
+    qid = input()
+
+    if not validate(qid, int(Config.QID_MODULUS), int(Config.QID_FACTOR)):
+        raise ValueError('ERROR: Bad QID, checksum invalid')
+
+    qid_check_response = requests.get(f'http://{Config.CASEAPI_HOST}:{Config.CASEAPI_PORT}/qids/{qid}')
+    if qid_check_response.status_code != 404:
+        raise ValueError('ERROR: This QID already exists in the uac qid link table')
+
+    print('What is the UAC?')
+    uac = input()
+    if len(uac) != 16:
+        raise ValueError('Bad UAC, wrong length. Message not sent.')
+
+    if not uac.isalnum():
+        raise ValueError('Bad UAC, bad characters. Message not sent.')
+
+    check_uac_qid_pair_exist(uac, qid)
+
+    print('Whats the CaseID?')
+    case_id = input()
+
+    case_check_response = requests.get(f'http://{Config.CASEAPI_HOST}:{Config.CASEAPI_PORT}/cases/{case_id}')
+
+    try:
+        case_check_response.raise_for_status()
+    except HTTPError:
+        raise ValueError('Error: invalid or non-existent Case ID. Message not sent.')
+
+    message = {
+        "event": {
+            "type": "RM_UAC_CREATED",
+            "source": "RM",
+            "channel": "RM",
+            "dateTime": datetime.utcnow().isoformat() + 'Z',
+            "transactionId": str(uuid.uuid4())
+        },
+        "payload": {
+            "uacQidCreated": {
+                "caseId": case_id,
+                "qid": qid,
+                "uac": uac
+            }
+        }
+    }
+
+    print('Here is the generated message:')
+    print(json.dumps(message, sort_keys=True, indent=4))
+    print('Do you want to publish it?')
+    do_publish = input()
+
+    if do_publish != 'yes':
```

Comment:
Might be worth saying exactly what the user needs to type here (i.e.. a `y` won't work here like above)

Reply:
Agreed it would make sense but leaving it for now since it's the same across all the message makers